backend/rest_api_service/main.py

```diff
@@ -62,3 +64,38 @@ async def get_all_participants(response: Response, db: Session = Depends(get_db)
     except Exception as e:
         response.status_code = 500
         return {"message": f"An error occured: {e}"}
+    
+
+@app.get("/login/{username}", tags=['User'], status_code=status.HTTP_200_OK)
+async def login(username: str, password: str, response: Response, db: Session = Depends(get_db)):
+    try: 
+        user = await crud.get_user_by_username(db, username)
```

Comment:
Trzeba dodać sprawdzenie, czy user nie jest None, bo jeśli jest to w potem wyrzuci NoneType jak robimy `user.password_hash`